app/subscriber/src/features/settings/MyAccountSettings.tsx

```diff
@@ -0,0 +1,76 @@
+import React from 'react';
+import { FaEnvelope, FaToggleOff, FaToggleOn, FaUmbrellaBeach } from 'react-icons/fa';
+import { toast } from 'react-toastify';
+import { useUsers } from 'store/hooks';
+import { useProfileStore } from 'store/slices';
+import { ISubscriberUserModel, ToggleButton } from 'tno-core';
+
+import * as styled from './styled';
+
+const MyAccountSettings = () => {
+  const { updateUser } = useUsers();
+  const [{ profile, impersonate }, { storeMyProfile, storeImpersonate }] = useProfileStore();
+  const isVacationMode: boolean = !!impersonate
+    ? impersonate?.preferences?.isVacationMode ?? false
+    : profile?.preferences?.isVacationMode ?? false;
+
+  const toggleVacationMode = React.useCallback(async () => {
+    if (!profile) {
+      toast.error('User information is missing. Please try again later');
+      return;
+    }
+    const baseProfile = impersonate ?? profile;
+    const createUser = (): ISubscriberUserModel => {
+      // use impersonate if it exists, otherwise use profile
+      return {
+        ...baseProfile,
+        preferences: {
+          ...baseProfile.preferences,
+          isVacationMode: !isVacationMode,
+        },
+      };
+    };
+    const user = createUser();
+
+    try {
+      !!impersonate ? storeImpersonate(user) : storeMyProfile(user);
+      await updateUser(user, !!impersonate);
+      toast.success('Vacation mode has successfully been updated.');
+    } catch (error) {
+      // Handle the error, if needed
+      console.error('Failed to update user:', error);
+    }
+  }, [profile, impersonate, isVacationMode, storeImpersonate, storeMyProfile, updateUser]);
```

Comment:
Move `profile`, `impersonate` and `isVacationMode` to parameters in the function will reduce the rerendering required to keep this function in sync with the variables.  You can just call the function like this `toggleVacationMode(impersonate ?? profile, !isVacationMode)`